llvm/lib/CodeGen/MachineLICM.cpp

```diff
@@ -391,9 +390,8 @@ bool MachineLICMBase::runOnMachineFunction(MachineFunction &MF) {
     else {
       // CSEMap is initialized for loop header when the first instruction is
       // being hoisted.
-      MachineDomTreeNode *N = DT->getNode(CurLoop->getHeader());
       FirstInLoop = true;
-      HoistOutOfLoop(N, CurLoop, CurPreheader);
```

Comment:
`N` is no longer a node in `DT` when we split critical edges eagerly in failed tests before.